tests/unit/dataset.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { mongoResultsToDataset, MongoResults } from '@/lib/dataset';
+
+describe('Dataset helper tests', () => {
+  it('should be able to handle empty mongo results', () => {
+    const mongoResults: MongoResults = [];
+    const dataset = mongoResultsToDataset(mongoResults);
+    expect(dataset.columns).toEqual([]);
+    expect(dataset.data).toEqual([]);
+  });
+
+  it('should be able to convert homegeneous mongo results', () => {
+    const mongoResults: MongoResults = [
+      { col1: 'foo', col2: 42, col3: true },
+      { col1: 'bar', col2: 7, col3: false },
+    ];
+    const dataset = mongoResultsToDataset(mongoResults);
+    expect(dataset.columns).toEqual([{ name: 'col1' }, { name: 'col2' }, { name: 'col3' }]);
+    expect(dataset.data).toEqual([['foo', 42, true], ['bar', 7, false]]);
+  });
+
+  it('should be able to convert heterogeneous mongo results', () => {
+    const mongoResults: MongoResults = [
+      { col1: 'foo', col3: true },
+      { col1: 'bar', col2: 7, col4: '?' },
+    ];
+    const dataset = mongoResultsToDataset(mongoResults);
+    expect(dataset.columns).toEqual([
+      { name: 'col1' },
+      { name: 'col2' },
+      { name: 'col3' },
+      { name: 'col4' },
+    ]);
+    expect(dataset.data).toEqual([['foo', undefined, true, undefined], ['bar', 7, undefined, '?']]);
```

Comment:
Ok nice one: so we are guaranteed that all rows will always be arrays of same length, with null values represented as `undefined`=> maybe we can document that clearly in the function implemented in dataset.ts ?

Reply:
Indeed

Reply:
I've eventually decided to go for `null` since I don't see the point of differentiating `null` values (i.e. specified but null) and `undefined` values.